src/Orbit.cpp

```diff
@@ -173,6 +183,19 @@ vector3d Orbit::OrbitalPosAtTime(double t) const
 	return m_orient * vector3d(-cos_v * r, sin_v * r, 0);
 }
 
+vector3d Orbit::OrbitalDisplacementAtTime(double t, double dt) const
+{
+	if (is_zero_general(m_semiMajorAxis)) return m_positionForStaticBody;
+	double cos_v_t1, sin_v_t1, r1, M1 = MeanAnomalyAtTime(t);
+	double cos_v_t2, sin_v_t2, r2, M2 = MeanAnomalyDiff(M1, dt);
+	calc_position_from_mean_anomaly(M1, m_eccentricity, m_semiMajorAxis, cos_v_t1, sin_v_t1, &r1);
+	calc_position_from_mean_anomaly(M2, m_eccentricity, m_semiMajorAxis, cos_v_t2, sin_v_t2, &r2);
```

Comment:
This is doing a lot of extra work for almost no gain. You compute the delta between a previously-known positition (calculated at Frame.cpp line 490) by re-deriving the position, then computing the future position to take the delta relative to the current position. No precision is saved in this function *at all* because you're still adding `2pi * dt / period` to a value many (likely >1000) multiples of Pi.

Because Mean Anomaly is relative to the most recent periapse ([page 9](https://control.asu.edu/Classes/MAE462/462Lecture04.pdf)), taking `frac(t / Period())` produces a much more accurate result for `MeanAnomalyAtTime`.

OrbitalDisplacementAtTime should additionally take in a `pos1` variable rather than re-computing it inline.